src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SingleSourceStageLogic.cs

```diff
@@ -22,67 +22,53 @@ internal class SingleSourceStageLogic<K, V, TMessage> : BaseSingleSourceLogic<K,
-                    ConsumerActor.Tell(new KafkaConsumerActorMetadata.Internal.SubscribePattern(topicSubscriptionPattern.TopicPattern), SourceActor.Ref);
-                    break;
-                case IManualSubscription manualSubscription:
-                    ConfigureManualSubscription(manualSubscription);
-                    break;
-                default:
-                    throw new NotSupportedException();
+                var strPart = (_settings.Properties.TryGetValue("client.id", out var clientId)) ?
+                    $"client-{_settings.GroupId}-{clientId}" : $"client-{_settings.GroupId}";
+                return Akka.Event.LogSource.Create(strPart, GetType());
             }
         }
 
         /// <inheritdoc />
         protected override IActorRef CreateConsumerActor()
         {
-            var partitionsAssignedHandler = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsAssigned);
-            var partitionsRevokedHandler = GetAsyncCallback<IEnumerable<TopicPartitionOffset>>(PartitionsRevoked);
-            var partitionsLostHandler = GetAsyncCallback<IEnumerable<TopicPartitionOffset>>(PartitionsLost);
-
-            IPartitionEventHandler internalHandler = new PartitionEventHandlers.AsyncCallbacks(partitionsAssignedHandler, partitionsRevokedHandler, partitionsLostHandler);
-
-            // If custom partition events handler specified - add it to the chain
-            var eventHandler = _subscription is IAutoSubscription autoSubscription && autoSubscription.PartitionEventsHandler.HasValue
-                ? new PartitionEventHandlers.Chain(autoSubscription.PartitionEventsHandler.Value, internalHandler)
-                : internalHandler;
-
-            IStatisticsHandler statisticsHandler = _subscription.StatisticsHandler.HasValue
-                ? _subscription.StatisticsHandler.Value
-                : new StatisticsHandlers.Empty();
-
-            // This allows to override partition events handling by subclasses
-            eventHandler = AddToPartitionAssignmentHandler(eventHandler);
+            IStatisticsHandler statisticsHandler = Subscription.StatisticsHandler.HasValue
+                ? Subscription.StatisticsHandler.Value
+                :  StatisticsHandlers.Empty.Instance;
             
-            if (!(Materializer is ActorMaterializer actorMaterializer))
+            if (Materializer is not ActorMaterializer actorMaterializer)
                 throw new ArgumentException($"Expected {typeof(ActorMaterializer)} but got {Materializer.GetType()}");
             
             var extendedActorSystem = actorMaterializer.System.AsInstanceOf<ExtendedActorSystem>();
-            var actor = extendedActorSystem.SystemActorOf(KafkaConsumerActorMetadata.GetProps(SourceActor.Ref, _settings, Decider, eventHandler, statisticsHandler),
+            var actor = extendedActorSystem.SystemActorOf(KafkaConsumerActorMetadata.GetProps(SourceActor.Ref, _settings, Decider, statisticsHandler),
```

Comment:
Stats handler still gets passed in via CTOR since it's not really re-configurable the same way the `IPartitionHandler`s are.